### PR TITLE
BGDIINF_SB-1478: field type of Asset has to be required and validated

### PR DESCRIPTION
**Pull request by rebert**
Putting all validators into a sepperate file

### CODE REVIEW

**Comment by rebert** [comment]
@here It seems that the validator _validate_item_properties_datetimes_dependencies_ is not being used any more. Is it ok, when I remove it?

**Comment by hansmannj** [comment]
> @here It seems that the validator _validate_item_properties_datetimes_dependencies_ is not being used any more. Is it ok, when I remove it?

From my understanding it is still in use. It is called from within `validate_item_properties_datetimes()` which is called from within the item's `clean()`. So I'd suggest to leave it as is.

**Comment by hansmannj** [comment]
P.S. I guess the ticket still says, that `type` (called `media_type` in `models.py`) should be set to required. This has already been done ;-)

**Comment by rebert** [comment]
@here:
- I did not use the field type Choices, as the media types are sometimes with, sometimes without space
- Even though it has defined diffined:
  `image/tiff; application=geotiff; profile=cloud-optimized` 
  I left it:
  `image/tiff; application=geotiff; profile=cloud-optimize` (without **d**)

File: app/stac_api/validators.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+from stac_api.utils import fromisoformat
+
+logger = logging.getLogger(__name__)
+
+MEDIA_TYPES = [
+    ('application/x.asc+zip', 'Zipped ASCII (.asc)', '.zip'),
+    ('application/x.asc-grid+zip', 'Zipped ASCII-GRID (.xyz)', '.zip'),
+    ('application/x.e00+zip 	Zipped', 'e00', '.zip'),
```
Comment (ltshb) [inline app/stac_api/validators.py:15]: Remove tab in media type

File: app/stac_api/validators.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+def validate_item_properties_datetimes_dependencies(
+    properties_datetime, properties_start_datetime, properties_end_datetime
+):
+    '''
+    Validate the dependencies between the Item datetimes properties
+	This makes sure that either only the properties.datetime is set or
+	both properties.start_datetime and properties.end_datetime
+	Raises:
+		django.core.exceptions.ValidationError
```
Comment (ltshb) [inline app/stac_api/validators.py:141]: TABS ! only space in code.

File: app/stac_api/models.py [hunk 3]
```diff
@@ -656,6 +531,12 @@ def from_db(cls, db, field_names, values):
     def __str__(self):
         return self.name
 
+    def clean(self):
+        # as the media type sometimes is with space, sometimes without
+        # it will alway be stored without
+        if self.media_type:
+            self.media_type = self.media_type.replace(' ', '')
```
Comment (ltshb) [inline app/stac_api/models.py:538]: This is a field only clean step and should be placed in `clean_media_type()` method (see https://docs.djangoproject.com/en/3.1/ref/forms/validation/#form-and-field-validation)

**Review by ltshb — CHANGES_REQUESTED** [review]
Good only issue regarding space and TAB.

In my opinion we have to decide the separator for the media type:

1. SEMI-COLON + SPACE (e.g. `; `)
1. SEMI-COLON (e.g. `;`)

Then the media type list should be done accordingly.

Then the question do we want to support extra space separator and automatically strip them ? I would not, in the admin page the field is a choice field anyway and the API will be used by machine with correct type string. @boecklic what's your opinion ?

File: app/stac_api/validators.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
+import logging
```
Comment (hansmannj) [inline app/stac_api/validators.py:1]: :+1:
Good idea to source-out the validators to a separate file :smile:

File: app/stac_api/validators.py (hunk 1 above)
Comment (hansmannj) [reply app/stac_api/validators.py:15]: They use `; ` (semi-colon + space) here: https://github.com/radiantearth/stac-spec/blob/v0.9.0/item-spec/item-spec.md#media-types.
So I'd suggests to stick to that, unless it causes problems with the choice field. In that case I'd drop the extra space and the world will still keep on turning ;-)

Comment (hansmannj) [reply app/stac_api/validators.py:15]: P.S. and they call it `cloud-optimized` (with d)

**Review by hansmannj — APPROVED** [review]
Thanks, LGTM :smiley:

**Pull request merged**